wpilibc/src/main/native/include/frc/logging/CSVLogFile.h

```diff
@@ -0,0 +1,69 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <chrono>
+#include <sstream>
+
+#include <wpi/StringRef.h>
+
+#include "frc/logging/LogFile.h"
+
+namespace frc {
+
+/**
+ * A CSVLogFile writes to a csv file
+ *
+ * For the CSVLogFile to write log informations, you must call Log()
+ * periodically.
+ */
+class CSVLogFile {
+ public:
+  /**
+   * Instantiate a LogFile passing in its prefix and its extension.
+   *
+   * If you want the file to be saved in a existing directory, you can add its
+   * path before the file prefix. Exemple : to save the file in a usb stick on
+   * the roborio ("/media/sda1/") : LogFile("/media/sda1/log").
+   *
+   * @param filePrefix The prefix of the LogFile.
+   */
+  template <typename ValueT, typename... ValuesT>
+  CSVLogFile(wpi::StringRef filePrefix, ValueT value, ValuesT... values)
+      : m_logFile(filePrefix, "csv") {
+    m_logFile.Log("Timestamp (ms),");
+    LogValues(value, values...);
+  }
+
+  template <typename ValueT, typename... ValuesT>
+  void Log(ValueT value, ValuesT... values) {
+    using namespace std::chrono;
+    auto timestamp =
+        duration_cast<milliseconds>(system_clock::now().time_since_epoch());
+    m_logFile.Log(std::to_string(timestamp.count()) + ",");
+
+    LogValues(value, values...);
+  }
+
+ private:
+  template <typename ValueT, typename... ValuesT>
+  void LogValues(ValueT value, ValuesT... values) {
+    std::stringstream stringConverter;
```

Comment:
Rather than using a stringstream, we need to handle string types specially. I just looked at how LibreOffice handles CSVs, and it includes quotes around strings when saving and leaves integers alone. When you open the file, it asks for the value delimiter (comma), and string delimiter (quote).

You can do this:
```cpp
if constexpr (std::is_convertible_v<Value, wpi::StringRef>) {
  m_logFile.Log(wpi::Twine("\"") + value + "\",");
} else {
  m_logFile.Log(std::to_string(value) + ",");
}
```
The first branch might need some tweaks to compile. `wpi::Twine` is in `<wpi/Twine.h>`. It's a way to efficiently concatenate strings. Idk if `LogFile::Log()` will need a `wpi::Twine` overload or not.